File: Back-End/src/main/java/com/boyslab/ads/dtos/response/reportingDebris/ReportingDebrisResponseDto.java
package com.boyslab.ads.dtos.response.reportingDebris;

import com.boyslab.ads.dtos.request.reportingDebris.ReportingDebrisDto;
import com.boyslab.ads.entities.ReportingDebris;

public record ReportingDebrisResponseDto(
        Integer id,
        String phoneNumber,
        String imageUrl,
        String city,
        String district,
        String neighbourhood,
        String street,
        String locationDescription,
        Boolean ekipVarMı
) {
    public static ReportingDebrisResponseDto convertToDto(ReportingDebris reportingDebris){
        return new ReportingDebrisResponseDto(
                reportingDebris.getId(),
                reportingDebris.getPhoneNumber(),
                reportingDebris.getImage(),
                reportingDebris.getCity(),
                reportingDebris.getDistrict(),
                reportingDebris.getNeighbourhood(),
                reportingDebris.getStreet(),
                reportingDebris.getLocationDescription(),
                reportingDebris.getEkipVarMı()
        );
    }
}

File: AfetProjesi/app/src/main/java/com/example/afetprojesi/dtos/responses/reportingDebris/ReportingDebrisResponseDto.kt
package com.example.afetprojesi.dtos.responses.reportingDebris

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class ReportingDebrisResponseDto(
    @Expose
    @SerializedName("id")
    var id:Int,

    @Expose
    @SerializedName("phoneNumber")
    var phoneNumber: String,

    @Expose
    @SerializedName("imageUrl")
    var imageUrl : String,

    @Expose
    @SerializedName("city")
    var city:String,

    @Expose
    @SerializedName("district")
    var district : String,
    @Expose
    @SerializedName("neighbourhood")
    var neighbourhood:String,

    @Expose
    @SerializedName("street")
    var street : String,

    @Expose
    @SerializedName("locationDescription")
    var locationDescription : String,

    @Expose
    @SerializedName("ekipVarmı")
    var ekipVarmi :String
)
